sliceutil/sliceutil.go

```diff
@@ -201,3 +211,17 @@ func Values[T comparable, N any](tt []T, fn func(T) N) []N {
 
 	return ret
 }
+
+// IsSubset returns if all elements of 'slice' are in 'set'
+func IsSubset[T comparable](slice, subset []T) bool {
+	subsetMap := make(map[T]bool, len(subset))
+	for _, v := range subset {
+		subsetMap[v] = true
+	}
+	for _, v := range slice {
+		if !subsetMap[v] {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
This func right now is flipped (checking if `slice` is in `subset` rather than the other way round), something like this might work better:

```suggestion
// IsSubset returns if all elements of 'slice' are in 'set'
func IsSubset[T comparable](set, subset []T) bool {
	setMap := make(map[T]bool, len(set))
	for _, v := range set {
		setMap[v] = true
	}
	for _, v := range subset {
		if !setMap[v] {
			return false
		}
	}
	return true
}
```

This is also why your tests are failing as the logic is basically flipped. Also even with the above change the last test would fail, however the expected value of `false` is incorrect as any set is considered a subset of itself.

Also considering this is working with sets, I'd move it to `sets.go` (and tests to `sets_test.go`) rather than the base `sliceutil.go` file.

Lastly, if you'd prefer a bit more readability/reuse over performance, you could simplify it to something like:
```
for _, v := range subset {
	if !Contains(set, v) {
		return false
	}
}
return true
```